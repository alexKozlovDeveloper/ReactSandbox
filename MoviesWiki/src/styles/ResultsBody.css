.body {
    width: 100%;
    margin: 0 auto;
    /* padding-top: 128px;
    padding-bottom: 35px; */
    background-image: url(http://hd-wallpapers.co/wp-content/uploads/walls/thumbs/Dark-Blue-1024x640.jpg);
    background-size: contain;
    min-width: 500px;
    min-height: calc(100vh - 163px);
}

.infocontainer {
    width: 100%;
    float: left;
}