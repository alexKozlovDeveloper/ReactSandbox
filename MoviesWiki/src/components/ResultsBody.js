import React, { Component } from "react";
import { connect } from 'react-redux'

import ResultsCount from "./Common/ResultsCount";
import ItemsContainer from "./Container/ItemsContainer";
import CustomOptionList from "./Common/CustomOptionList";
import Spinner from "./Common/Spinner";

import { updateSelectedItem, updateMovies, updateSortBy } from "../actions/index";

import styles from "../styles/ResultsBody.css"

class ResultsBody extends Component {
    constructor(props) {
        super(props);
    }

    componentDidMount() {
        if (this.props.isLoaded == false) {
            fetch("https://reactjs-cdp.herokuapp.com/movies?sortBy=id")
                .then(res => res.json())
                .then(
                    (result) => {
                        this.props.updateMoviesFunc(result);
                    },
                    (error) => {
                        // TODO: Process error
                    }
                )
        }
    }

    render() {

        if (this.props.isLoaded === false || this.props.movies == null) {
            return (
                <div className={styles.body}>
                    <Spinner></Spinner>
                </div>)
        }

        var selectedIndex = this.props.sortBy === 'release_date' ? "0" : "1";

        var sortBy = this.props.sortBy;

        function compare(a, b) {
            if (a[sortBy] < b[sortBy]) {
                return -1;
            }
            if (a[sortBy] > b[sortBy]) {
                return 1;
            }
            return 0;
        }

        let sortedMovies = this.props.movies.concat([]);
        sortedMovies.sort(compare);

        return (
            <div className={styles.body}>
                <div className={styles.infocontainer}>
                    <div className={styles.resultscountcontainer}>
                        <ResultsCount count={this.props.movies.length} title={this.props.config.resultsCountConfig.title} />
                    </div>
                    <div className={styles.customoptionlistcontainer}>
                        <CustomOptionList config={this.props.config.resultSortConfig} selectedIndex={selectedIndex} updateFunc={this.props.updateSortByFunc} />
                    </div>
                </div>
                <div>
                    <ItemsContainer items={sortedMovies} itemsPerRow={this.props.config.itemsPerRow} />
                </div>
            </div>
        );
    }
}

function mapDispatchToProps(dispatch) {
    return {
        updateMoviesFunc: (movies) => {
            dispatch(updateMovies(movies.data))
        },
        updateSortByFunc: (sortBy) => {
            dispatch(updateSortBy(sortBy))
        },
        updateSelectedItemFunc: (item) => {
            dispatch(updateSelectedItem(item))
        }
    }
}

function mapStateToProps(state) {
    const { movies, isLoaded, sortBy } = state.moviesReducer;

    return {
        movies: movies,
        isLoaded: isLoaded,
        sortBy: sortBy
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(ResultsBody);