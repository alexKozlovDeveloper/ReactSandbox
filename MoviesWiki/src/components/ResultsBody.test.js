import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import rootReducer from '../reducers';

import ResultsBody from './ResultsBody';

configure({ adapter: new Adapter() });

test('ResultsBody rendering test', () => {
  const resultsBodyConfig = {
    itemsPerRow: 4,
    resultSortConfig: {
      title: 'Sort by',
      selectedIndex: '1',
      options: ['Release date', 'Rating'],
    },
    resultsCountConfig: {
      title: 'movies found',
    },
  };

  const moviesList = [{
    title: 'Interstellar',
    releaseDate: '2014',
    image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
    genre: 'action',
    rating: 7.2,
    duration: 145,
    description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
  }, {
    title: 'Interstellar 2',
    releaseDate: '2015',
    image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
    genre: 'action2',
    rating: 7.2,
    duration: 145,
    description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
  }, {
    title: 'Interstellar 3',
    releaseDate: '2016',
    image: 'https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg',
    genre: 'action 3',
    rating: 7.2,
    duration: 145,
    description: 'is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.',
  }];

  const defaultStore = {
    moviesReducer: {
      movies: moviesList,
      isLoaded: true,
      error: null,
      sortBy: 'vote_average',
    },
  };


  const store = createStore(rootReducer, defaultStore);

  const component = shallow(<Provider store={store}><ResultsBody items={moviesList} config={resultsBodyConfig} /></Provider>);
  expect(shallowToJson(component)).toMatchSnapshot();
});
