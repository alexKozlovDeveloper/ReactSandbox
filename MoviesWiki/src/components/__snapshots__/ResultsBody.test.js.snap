// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsBody rendering test 1`] = `
<div
  className="body"
>
  <div
    className="infocontainer"
  >
    <div
      className="resultscountcontainer"
    >
      <ResultsCount
        count={3}
        title="movies found"
      />
    </div>
    <div
      className="customoptionlistcontainer"
    >
      <CustomOptionList
        config={
          Object {
            "options": Array [
              "Release date",
              "Rating",
            ],
            "selectedIndex": "1",
            "title": "Sort by",
          }
        }
      />
    </div>
  </div>
  <div>
    <ItemsContainer
      items={
        Array [
          Object {
            "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
            "duration": 145,
            "genre": "action",
            "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            "rating": 7.2,
            "releaseDate": "2014",
            "title": "Interstellar",
          },
          Object {
            "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
            "duration": 145,
            "genre": "action2",
            "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            "rating": 7.2,
            "releaseDate": "2015",
            "title": "Interstellar 2",
          },
          Object {
            "description": "is a 2014 British-American epic science fiction film directed and produced by Christopher Nolan. ... Set in a dystopian future where humanity is struggling to survive, the film follows a group of astronauts who travel through a wormhole near Saturn in search of a new home for humanity.",
            "duration": 145,
            "genre": "action 3",
            "image": "https://m.media-amazon.com/images/M/MV5BZjdkOTU3MDktN2IxOS00OGEyLWFmMjktY2FiMmZkNWIyODZiXkEyXkFqcGdeQXVyMTMxODk2OTU@._V1_SY1000_SX675_AL_.jpg",
            "rating": 7.2,
            "releaseDate": "2016",
            "title": "Interstellar 3",
          },
        ]
      }
      itemsPerRow={4}
    />
  </div>
</div>
`;
